drivers/eeprom/eeprom_fm25w256.c

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright (c) 2021 Caspar Friedrich <c.s.w.friedrich@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <devicetree.h>
+#include <devicetree/gpio.h>
+#include <drivers/eeprom.h>
+#include <drivers/gpio.h>
+#include <drivers/spi.h>
+#include <logging/log.h>
+
+#define DT_DRV_COMPAT cypress_fm25w256
+
+#define LEVEL POST_KERNEL
+#define PRIORITY CONFIG_SPI_INIT_PRIORITY
+
+// 256-Kbit ferroelectric random access memory logically organized as 32K × 8 bit
+#define SIZE KB(32)
+#define WORD_SIZE 8
+
+LOG_MODULE_REGISTER(fm25w256, CONFIG_EEPROM_LOG_LEVEL);
+
+// Opcode Commands
+enum {
+	OPCODE_WREN = 0b00000110,
+	OPCODE_WRDI = 0b00000100,
+	OPCODE_RDSR = 0b00000101,
+	OPCODE_WRSR = 0b00000001,
+	OPCODE_READ = 0b00000011,
+	OPCODE_WRITE = 0b00000010,
+};
```

Comment:
Any particular reason for specifying these in binary? Most other drivers would use hex.

Reply:
The datasheet lists them as binary. I kept it because of consistency.